### ISSUE TITLE
runtime error: left shift of 162 by 24 places cannot be represented in type 'int'

### ISSUE THREAD

**Issue by merbanan** [merbanan/rtl_433#1193]
rtl_433 -R 130 -s 250k -Y 1 -r rtl_433_tests/tests/IKEA_Sparsnas/01/*.cu8

gives

Test mode active. Reading samples from file: rtl_433_tests/tests/IKEA_Sparsnas/01/g001_867.95M_250k.cu8
/home/benjamin/projects/rtl_433/src/devices/ikea_sparsnas.c:116:53: runtime error: left shift of 162 by 24 places cannot be represented in type 'int'

(k0 << 24) -> (((uint32_t)k0 << 24) gets rid of the error message but is it the correct way to do it? Is changing the k0 type better?

**Comment by zuckschwerdt** [comment]
Yes, that's the right change. We have that in all other places like this. C integer promotion rules give us int (int32_t) for the right side, we need to force unsigned (uint32_t) if we use the uppermost bits. Thus k0 should trigger that. I'll push the change.
